packages/zosmf/__tests__/__system__/api/methods/ListDefinedSystems.system.test.ts

```diff
@@ -0,0 +1,154 @@
+                error = err;
+                Imperative.console.info("Error: " + inspect(error));
+            }
+
+            expect(error).toBeFalsy();
+            expect(response).toBeTruthy();
+            expect(JSON.stringify(response)).toContain("items");
+        });
+    });
+
+    describe("Failure scenarios", () => {
+        it("should return with proper message for invalid session", async () => {
+            let error;
+            let response;
+
+            try {
+                response = await ListDefinedSystems.listDefinedSystems(undefined);
+            } catch (err) {
+                error = err;
+                Imperative.console.info("Error: " + inspect(error));
+            }
+
+            expect(error).toBeTruthy();
+            expect(response).toBeFalsy();
+            expect(error.message).toContain(ZosmfMessages.missingSession.message);
+        });
+
+        it("should return with proper message for invalid hostname", async () => {
+            const badHostName = "badHost";
+            const badSession = new Session({
+                user: defaultSystem.zosmf.user,
+                password: defaultSystem.zosmf.pass,
+                hostname: badHostName,
+                port: defaultSystem.zosmf.port,
+                type: "basic",
+                rejectUnauthorized: defaultSystem.zosmf.rejectUnauthorized,
+            });
+
+            let error;
+            let response;
+
+            try {
+                response = await ListDefinedSystems.listDefinedSystems(badSession);
+            } catch (err) {
+                error = err;
+                Imperative.console.info("Error: " + inspect(error));
+            }
+
+            expect(error).toBeTruthy();
+            expect(response).toBeFalsy();
+            expect(error.message).toContain(`Error: getaddrinfo ENOTFOUND ${badHostName}`);
+        });
+
+        it("should return with proper message for invalid port", async () => {
+            const badPort = 9999;
+            const badSession = new Session({
+                user: defaultSystem.zosmf.user,
+                password: defaultSystem.zosmf.pass,
+                hostname: defaultSystem.zosmf.host,
+                port: badPort,
+                type: "basic",
+                rejectUnauthorized: defaultSystem.zosmf.rejectUnauthorized,
+            });
+
+            let error;
+            let response;
+
+            try {
+                response = await ListDefinedSystems.listDefinedSystems(badSession);
+            } catch (err) {
+                error = err;
+                Imperative.console.info("Error: " + inspect(error));
+            }
+
+            expect(error).toBeTruthy();
+            expect(response).toBeFalsy();
+            expect(error.message).toContain(`Error: connect ECONNREFUSED`);
+            expect(error.message).toContain(badPort);
+        });
+
+        it("should return with proper message for rejectUnauthorized = true", async () => {
```

Comment:
This test won't necessarily pass if the system has non-self-signed certificates. Reject unuathorized=true could be a valid setting and won't necessarily produce an error. I would recommend removing this test.